### DISCUSSION TITLE
Feature Request: CREATE RANDOM STREAM add support for fractional EPS (slow events)

### DISCUSSION THREAD

**Discussion by mgb-git** [timeplus-io/proton#292] [Ideas]
CREATE RANDOM STREAM supports and EPS setting for events generated per second. But a fractional EPS (i.e., less than one per second; e.g. one event every 5 seconds, or EPS=0.2) is interpreted as EPS=0, no limit. For manual testing purposes, especially where time is important, having slower synthetic data is useful. I'm looking to validate some ASOF JOIN logic using time and this would help.

**Comment by yokofly** [comment]
shall be merged in https://github.com/timeplus-io/proton/pull/507
